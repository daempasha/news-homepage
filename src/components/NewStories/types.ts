export interface NewStoriesProps {
    data: { title: string, description: string }[]
}