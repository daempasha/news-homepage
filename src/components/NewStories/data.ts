export default [
    {
        title: "Hydrogen Vs Electric Cars",
        description: "Will hydrogen-fueled cars ever catch up to EVs?"
    },
    {
        title: "The Downsides of AI Artistry",
        description: "What are the possible adverse affects of on-demand AI image generation?"
    },
    {
        title: "Is VC Funding Drying Up",
        description: "Private funding by VC firms is down 50% YOY. We take a look at what that means."
    }
]