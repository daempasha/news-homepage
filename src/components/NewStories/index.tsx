import { NewStoriesProps } from "./types";

const NewStories: React.FC<NewStoriesProps> = ({ data }): React.ReactElement => {
    return <div className="flex-grow flex-shrink-0 md:w-96 bg-black p-5">
        <h1 className="text-4xl font-bold text-orange-300 mb-5">New</h1>
        {data.map((row, index) => (
            <>
                <div className="text-white my-5 md:my-10">
                    <a href="#" className="text-xl font-bold cursor-pointer hover:text-orange-500">{row.title}</a>
                    <p className="text-sm my-2 text-gray-400">{row.description}</p>

                </div>
                {(index < data.length - 1) && <div className="border-b-2 border-gray-600" />}
            </>
        ))}
    </div>
}

export default NewStories;