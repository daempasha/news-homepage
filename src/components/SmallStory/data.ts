import revivingRetroPcImage from "@assets/image-retro-pcs.jpg";
import imageGamingGrowthImage from "@assets/image-gaming-growth.jpg";
import topLaptopsImage from "@assets/image-top-laptops.jpg";

export default [
    {
        image: revivingRetroPcImage,
        title: "Reviving Retro PCs",
        description: "What happens when old PCs are given modern upgrades?"
    },
    {
        image: topLaptopsImage,
        title: "Top 10 Laptops of 2022",
        description: "Our best picks for various needs and budgets."
    },
    {
        image: imageGamingGrowthImage,
        title: "The Growth of Gaming",
        description: "How the pandemic has sparked fresh opportunities."
    }
]