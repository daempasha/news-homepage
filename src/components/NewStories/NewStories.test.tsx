import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import NewStories from '.';
import data from './data';





describe('<NewStories />', () => {
    it('renders correctly', () => {
        const newStories = render(<NewStories {...data} />)
        expect(newStories).toMatchSnapshot()
    });

});