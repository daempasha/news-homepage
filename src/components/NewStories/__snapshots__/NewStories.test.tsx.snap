// Vitest Snapshot v1

exports[`<NewStories /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-grow flex-shrink-0 md:w-96 bg-black p-5"
      >
        <h1
          class="text-4xl font-bold text-orange-300 mb-5"
        >
          New
        </h1>
        <div
          class="text-white my-5 md:my-10"
        >
          <a
            class="text-xl font-bold cursor-pointer hover:text-orange-500"
            href="#"
          >
            Hydrogen Vs Electric Cars
          </a>
          <p
            class="text-sm my-2 text-gray-400"
          >
            Will hydrogen-fueled cars ever catch up to EVs?
          </p>
        </div>
        <div
          class="border-b-2 border-gray-600"
        />
        <div
          class="text-white my-5 md:my-10"
        >
          <a
            class="text-xl font-bold cursor-pointer hover:text-orange-500"
            href="#"
          >
            The Downsides of AI Artistry
          </a>
          <p
            class="text-sm my-2 text-gray-400"
          >
            What are the possible adverse affects of on-demand AI image generation?
          </p>
        </div>
        <div
          class="border-b-2 border-gray-600"
        />
        <div
          class="text-white my-5 md:my-10"
        >
          <a
            class="text-xl font-bold cursor-pointer hover:text-orange-500"
            href="#"
          >
            Is VC Funding Drying Up
          </a>
          <p
            class="text-sm my-2 text-gray-400"
          >
            Private funding by VC firms is down 50% YOY. We take a look at what that means.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-grow flex-shrink-0 md:w-96 bg-black p-5"
    >
      <h1
        class="text-4xl font-bold text-orange-300 mb-5"
      >
        New
      </h1>
      <div
        class="text-white my-5 md:my-10"
      >
        <a
          class="text-xl font-bold cursor-pointer hover:text-orange-500"
          href="#"
        >
          Hydrogen Vs Electric Cars
        </a>
        <p
          class="text-sm my-2 text-gray-400"
        >
          Will hydrogen-fueled cars ever catch up to EVs?
        </p>
      </div>
      <div
        class="border-b-2 border-gray-600"
      />
      <div
        class="text-white my-5 md:my-10"
      >
        <a
          class="text-xl font-bold cursor-pointer hover:text-orange-500"
          href="#"
        >
          The Downsides of AI Artistry
        </a>
        <p
          class="text-sm my-2 text-gray-400"
        >
          What are the possible adverse affects of on-demand AI image generation?
        </p>
      </div>
      <div
        class="border-b-2 border-gray-600"
      />
      <div
        class="text-white my-5 md:my-10"
      >
        <a
          class="text-xl font-bold cursor-pointer hover:text-orange-500"
          href="#"
        >
          Is VC Funding Drying Up
        </a>
        <p
          class="text-sm my-2 text-gray-400"
        >
          Private funding by VC firms is down 50% YOY. We take a look at what that means.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
